src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1568,8 +1568,7 @@ module T = struct
         (*There's enough work. Check if they satisfy other constraints*)
         Resources.budget_sufficient resources
       then
-        if Resources.space_constraint_satisfied resources then (resources, log)
-        else if Resources.worked_more ~constraint_constants resources then
+        if Resources.worked_more ~constraint_constants resources then
```

Comment:
Why is this included?

Reply:
I wanted to test a build with https://github.com/MinaProtocol/mina/pull/8098 included and I was expecting it to land in master before this finalfinal2 branch landed, but it hasn't yet. I just reverted the merge commit from 8098 so there are no longer .ml changes on this branch.